decisions/DR007-Hierarchical-or-compositional-traits-for-specifications.md

```diff
@@ -0,0 +1,361 @@
+degree that these classifications should be enforced are discussed in the options below.
+
+The current design of the OpenAssetIO specification system comes from prior work, where it was
+successfully employed in the development of real-world editorial and post-production workflows
+connecting commercially available DCCs tools and asset management systems.
+
+## Relevant data
+
+There are a few challenges to defining specifications. The types of entities manipulated by
+Managers continually grow over time as new software and file formats and new ways of creating
+images and scenes are added. Specifications have to continue to be able to keep their guarantees
+about their types while new guarantees are added. This makes it easier for DCCs, their developers,
+and users of the pipeline to be able to add to the types available without losing shared behaviour
+between specifications - both across DCCs and shared between the OpenAssetIO community. These types
+of entities, relationships and locales can't be counted on to be exactly the same, but defining the
+similarities between them so that they can fulfill reliable behaviours is important so that the
+specifications have sharable meaning.
+
+OpenAssetIO requires a mechanism capable of precisely describing entities, relationships and
+locales with the following criteria:
+
+Hierarchical or Composable (to support coercion of custom definitions to well-known behavioural
+traits) Simple in structure (to simple data types for interop/serialisation) Robust (compile-time
+checks for programming errors - property access or invalid method arguments) Discoverable (runtime
+introspection/help) Definable out of core (by host integrations/managers)
+
+## Options considered
+
+### Option 1
+
+Hierarchical specifications with a single parent per child. We assume that every child fulfills the
+contract of its parent. For example, that all alembic files are a geometry cache, and that all
+geometry caches might have coordinate system based data, and that all geometry caches are a type of
+binary blob. This could be described in a specification: entity. blob.geometry._cache.alembic.
+Another example could be an exr render, which could be a type of image, which could also be a type
+of blob binary data, represented as: entity.blob.image.exr .
+
+#### Pros
+
+- only one way to represent the specification
+- automatic fall back mechanism through inheritance
+- proven in real-world, working scenarios
+
+#### Cons
+
+- difficult to insert new specifications in between types, requires updating the child type
+  identifier of the new insert
+- requires knowledge of as many possible entities as possible to be efficiently taxonomied into an
+  efficient and useful hierarchy
+- sharing behaviours between taxonomically unrelated types - for example a frame range applied to
+  both a geometry cache and an image - is difficult, it relies on duplicated property definitions
```

Comment:
[consider] Would using a 'shot' as an alternative to 'image' make things clearer? The current implementation shared frame range between geometry and images already via `file.geometry` and `file.image`, where frame range was defined on `file`.

Reply:
Yes, sounds good. Added in e7717b77781fa9927a5dd0e23863ca706d113557